src/index.js

```diff
@@ -33,8 +26,39 @@ items.forEach((item) => {
 
   input.addEventListener('change', () => {
     updateCompleted(item, input);
-    saveLocalStorage(items);
+    saveLocalStorage(itemsArray);
   });
 
   todos.appendChild(div);
+  const vertical = div.querySelector('.vertical');
+  const trash = div.querySelector('.trash');
+  const edit = div.querySelector('.edit');
+  edit.addEventListener('change', () => {
+    editTodo(itemsArray, div.id, edit.value);
+    saveLocalStorage(itemsArray);
+  });
+  vertical.addEventListener('click', () => {
+    vertical.style.display = 'none';
+    trash.style.display = 'block';
+  });
+  trash.addEventListener('click', () => {
+    div.remove();
+    itemsArray = itemsArray.filter((i) => removeTodos(i, div.id));
+    saveLocalStorage(itemsArray);
+  });
+});
+
+const form = document.getElementById('form-id');
+form.addEventListener('submit', (e) => {
```

Comment:
- [ ] After fixing the two issues, the `add` functionality still not working. I think it is because the `<input>` field does not have a `submit` button because the icon actually is a `<span>` tag and another reason is that the `<input>` field is not inside a `<form>` tag (and it is a [valid option to use `<input>` without `<form>`](https://www.w3.org/TR/html401/interact/forms.html#form-controls)), but I would strongly suggest wrapping the `<input>` field inside a form and use a `<button>` or `<input type="submit">` instead of the `<span>` tag :wink:
**Hint: That's a good reason to keep your `<input>` field inside a `<form>` tag because by default the submit `<button>` will belong to this form**
![image](https://user-images.githubusercontent.com/81522435/145217384-1dda125c-8262-4c08-b06c-0131fdb335e8.png)